atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -37,45 +38,58 @@ def get_config(workflow):
         return conf
 
 
-def get_koji(workflow):
-    return get_config(workflow)['koji']
+def get_value(workflow, name, fallback):
+    try:
+        return get_config(workflow).conf[name]
+    except KeyError:
+        if fallback != NO_FALLBACK:
+            return fallback
+        raise
+
+
+def get_koji(workflow, fallback=NO_FALLBACK):
+    return get_value(workflow, 'koji', fallback)
 
 
-def get_koji_session(workflow):
-    config = get_koji(workflow)
+def get_koji_session(workflow, fallback):
+    config = get_koji(workflow, fallback)
 
     from atomic_reactor.koji_util import create_koji_session
+
     auth_info = {
         "proxyuser": config['auth'].get('proxyuser'),
         "ssl_certs_dir": config['auth'].get('ssl_certs_dir'),
         "krb_principal": config['auth'].get('krb_principal'),
         "krb_keytab": config['auth'].get('krb_keytab_path')
     }
+
+    if bool(auth_info['krb_principal']) != bool(auth_info['krb_keytab']):
```

Comment:
Should this be in `get_koji` instead of `get_koji_session`?

Reply:
I can move it there